### PR TITLE
Enable secrets and configMaps encryption in etcd rest on the apiserver

### PR DESCRIPTION
**Pull request by charlieegan3**
**What this PR does / why we need it**:
This makes use of the EncryptionConfig file written to vault by vault helper.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Implemented following the guide here: https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/

**This requires the following changes in vault-helper** https://github.com/jetstack/vault-helper/pull/23

**Release note**:
```release-note
Enable secrets and configMaps encryption in etcd rest on the apiserver
```

### CODE REVIEW

**Comment by charlieegan3** [comment]
At a glance it looks like my API server isn't coming up and so all the tests are failing. I suspect that until the vault helper changes are in this will continue to fail.

File: puppet/modules/vault_client/files/vault-add-hostname.sh [hunk 1]
```diff
@@ -5,4 +5,11 @@ set -ex
...
 
-vault read -format=json test/pki/k8s/roles/kubelet | python -c "import socket, sys, json; v=json.load(sys.stdin); v=v['data']; k='allowed_domains'; d=v[k].split(','); d.append(socket.gethostname()); v[k] = ','.join(list(set(d))); print json.dumps(v)" | vault write test/pki/k8s/roles/kubelet -
+#Download jq binary
+if [ ! -x /bin/jq ]; then
+    curl -sL -o jq https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
+    mv jq /bin/jq
+    chmod +x /bin/jq
+fi
+
+vault read -format=json test/pki/k8s/roles/kubelet | jq ".data | .allowed_domains += [\"$(hostname)\"]" | vault write test/pki/k8s/roles/kubelet -
```
[the remark this answers is not in the document]
Comment (charlieegan3) [reply puppet/modules/vault_client/files/vault-add-hostname.sh]: This is the new python code:

```
python -c "import socket, sys, json; v=json.load(sys.stdin); v=v['data']; k='allowed_domains'; d=v[k]; d.append(socket.gethostname()); print json.dumps(v)"
```

**Comment by simonswine** [comment]
Ok this works perfect for me

/assign @charlieegan3 
/unassign

**Comment by charlieegan3** [comment]
:tada:

**Pull request merged**